apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/JaxRsOffsetMappingFactory.java

```diff
@@ -0,0 +1,168 @@
+                }
+                return Target.ForStackManipulation.of(value);
+            }
+        };
+    }
+
+    private String getTransactionAnnotationValueFromAnnotations(MethodDescription instrumentedMethod, TypeDescription instrumentedType) {
+        TransactionAnnotationValue transactionAnnotationValue = new TransactionAnnotationValue();
+        String methodName = instrumentedMethod.getName();
+        while (!"java.lang.Object".equals(instrumentedType.getCanonicalName())) {
+            getAnnotationValueFromAnnotationSource(transactionAnnotationValue, instrumentedType, true);
+            for (MethodDescription.InDefinedShape annotationMethod : instrumentedType.getDeclaredMethods().filter(named(methodName)).asDefined()) {
+                getAnnotationValueFromAnnotationSource(transactionAnnotationValue, annotationMethod, false);
+            }
+            findInInterfaces(transactionAnnotationValue, instrumentedType, methodName);
+            instrumentedType = instrumentedType.getSuperClass().asErasure();
+        }
+        return transactionAnnotationValue.buildTransactionName();
+    }
+
+    private void findInInterfaces(TransactionAnnotationValue transactionAnnotationValue, TypeDescription classTypeDescription, String methodName) {
+        TypeList interfaces = classTypeDescription.getInterfaces().asErasures();
+        for (int i = 0; transactionAnnotationValue.method == null && i < interfaces.size(); i++) {
+            TypeDescription interfaceDescription = interfaces.get(i);
+            getAnnotationValueFromAnnotationSource(transactionAnnotationValue, interfaceDescription, false);
+            for (MethodDescription.InDefinedShape annotationMethod : interfaceDescription.getDeclaredMethods().filter(named(methodName))) {
+                getAnnotationValueFromAnnotationSource(transactionAnnotationValue, annotationMethod, false);
+            }
+            findInInterfaces(transactionAnnotationValue, interfaceDescription, methodName);
+        }
+    }
+
+    public void getAnnotationValueFromAnnotationSource(TransactionAnnotationValue transactionAnnotationValue, AnnotationSource annotationSource, Boolean isSuperclass) {
+        for (TypeDescription classMethodTypeDescription : annotationSource.getDeclaredAnnotations().asTypeList()) {
+            String canonicalName = classMethodTypeDescription.getCanonicalName();
+            switch (canonicalName) {
+                case "javax.ws.rs.Path":
+                    for (MethodDescription.InDefinedShape annotationMethod : classMethodTypeDescription.getDeclaredMethods().filter(named("value"))) {
+                        Object pathValue = annotationSource.getDeclaredAnnotations().ofType(classMethodTypeDescription).getValue(annotationMethod).resolve();
+                        if (pathValue != null) {
+                            if (isSuperclass) {
+                                transactionAnnotationValue.prependToPath((String) pathValue);
```

Comment:
That seems a little strange to me. I'd structure the algorithm this way:
1) Find the first class or interface in the class hierarchy which is annotated with `@Path`. Store in `TransactionAnnotationValue.classLevelPath`. Remove trailing slashes and ensure that there's a leading slash.
2) Find the first method-level `@GET`, ... in the class hierarchy and store its `value()` in `TransactionAnnotationValue.method`
3) Find the first method-level `@Path` annotation in the class hierarchy and store in `TransactionAnnotationValue.methodLevelPath`. Remove trailing slashes and ensure that there's a leading slash.
4) Construct the final signature: `method + " " + classLevelPath + methodLevelPath`

Initialize both `TransactionAnnotationValue.classLevelPath` and `TransactionAnnotationValue.methodLevelPath` with `""`, as both are optional.

Reply:
@felixbarny please check last changes.
Should I differentiate getAnnotationValueFromAnnotationSource method ?